File: api/src/models/photo.js
const mongoose = require('mongoose');
const Web3 = require('web3');

const photoSchema = new mongoose.Schema({
    index: {
        type: Number,
        unique: true,
        required: true
    }, 
    imagePath: {
        type: String,
        required: true
    },
    travellerAddress: {
        type: String,
        unique: true,
        required: true
    },
    helperAddress: {
        type: String,
        unique: true,
        required: true
    },
    travellerManager: {
        type: String, 
        required: true,
        trim: true
    },
});

const Photo = mongoose.model('Photo', photoSchema);

module.exports = Photo;

File: mobile_apps/lib/models/photo.dart
class Photo {
  int index;
  String imagePath;
  String travellerAddress;
  String helperAddress;
  String travellerManager;

  Photo({
    this.index,
    this.imagePath,
    this.travellerAddress,
    this.helperAddress,
    this.travellerManager,
  });

  Photo.fromJson(Map<String, dynamic> json)
      : index = json['index'],
        imagePath = json['imagePath'],
        travellerAddress = json['travellerAddress'],
        helperAddress = json['helperAddress'],
        travellerManager = json['travellerManager'];

  Map<String, dynamic> toJson() => {
        'index': index,
        'imagePath': imagePath,
        'travellerAddress': travellerAddress,
        'helperAddress': helperAddress,
        'travellerManager': travellerManager,
      };
}
